src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -5,67 +5,302 @@
+    void doNamesMatch_Yes_Apostrophe_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("  adriaNno ", "d'onOfio ", " adriaNno", " d onofiO  "), firstAndLastNameFailedMsg("  adriaNno ", "d'onOfio ", " adriaNno", " d onofiO  "));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" adriaNno  ", " doNofio", "adriaNno ", "  d'onoFio "), firstAndLastNameFailedMsg(" adriaNno  ", " doNofio", "adriaNno ", "  d'onoFio "));
     }
 
     @Test
-    void doNamesMatch_Maybe_FirstLastSwap() {
-        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("John", "Doe", "Doe", "John"));
+    void doNamesMatch_Yes_Apostrophe_CaseInsensitive_IgnoreSpaces_Punctuation() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("  josepHine-mc-vitty ", "d'onOfio ", " josephiNe mc vItty", " d onofiO  "), firstAndLastNameFailedMsg("  josepHine-mc-vitty ", "d'onOfio ", " josephiNe mc vItty", " d onofiO  "));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" josepHine mc vitty  ", " doNofio", "Josephine-mc-vitty ", "  d'onoFio "), firstAndLastNameFailedMsg(" josepHine mc vitty  ", " doNofio", "Josephine-mc-vitty ", "  d'onoFio "));
     }
 
     @Test
-    void doNamesMatch_Maybe_FirstNameInitials() {
-        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("John", "Doe", "J", "Doe"));
-        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("J.", "Doe", "John", "Doe"));
+    void doNamesMatch_Maybe_Apostrophe_CaseInsensitive_IgnoreSpaces_Hyphenated() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("  josepHine-mc-vitty ", "d'onOfio ", " josephiNe vItty", " d onofiO  "), firstAndLastNameFailedMsg("  josepHine-mc-vitty ", "d'onOfio ", " josephiNe vItty", " d onofiO  "));
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch(" mc vitty  ", " doNofio", "Josephine-mc-vitty ", "  d'onoFio "), firstAndLastNameFailedMsg(" mc vitty  ", " doNofio", "Josephine-mc-vitty ", "  d'onoFio "));
     }
 
     @Test
-    void doNamesMatch_no_FirstNameInitialsDifferentLastName() {
-        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Adams", "J", "Doe"));
-        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("J.", "Doe", "John", "Adams"));
+    void doNamesMatch_No_Apostrophe_CaseInsensitive_IgnoreSpaces_DifferentFirstName() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("  josepHine-mc-vitty ", "d'onOfio ", " josepH", " d onofiO  "), firstAndLastNameFailedMsg("  josepHine-mc-vitty ", "d'onOfio ", " josepH", " d onofiO  "));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch(" joSeph  ", " doNofio", "Josephine-mc-vitty ", "  d'onoFio "), firstAndLastNameFailedMsg(" joSeph  ", " doNofio", "Josephine-mc-vitty ", "  d'onoFio "));
     }
 
     // TODO: Add other name tests
```

Comment:
Are there other tests you think we need to add? If not, we can remove this TODO.